test/generate.test.ts

```diff
@@ -210,26 +233,65 @@ describe('generate', function () {
       assert.equal((answerSet[1].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
     });
 
-    it('should enumerate correct scaletype for nominal field', () => {
+    it('should enumerate correct scaleType for ordinal field without timeunit', () => {
       const specQ = {
         mark: Mark.POINT,
         encodings: [
           {
             channel: Channel.X,
             scale: {type: {values: [ScaleType.ORDINAL, undefined]}},
-            field: 'N',
+            field: 'O',
             type: Type.ORDINAL
           }
         ]
       };
 
       const answerSet = generate(specQ, schema, stats);
+      assert.equal(answerSet.length, 1);
+      assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
+    });
+
+    it('should enumerate correct scaleType for nominal field without timeunit', () => {
+      const specQ = {
+        mark: Mark.POINT,
+        encodings: [
+          {
+            channel: Channel.X,
+            scale: {type: {values: [ScaleType.ORDINAL, undefined]}},
+            field: 'N',
+            type: Type.NOMINAL
+          }
+        ]
+      };
+
+      const answerSet = generate(specQ, schema, stats);
+      assert.equal(answerSet.length, 1);
+      assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
+    });
+  });
+
+  it('should enumerate correct scaleType for nominal field with timeunit', () => {
```

Comment:
"should enumerate correct scaleType for nominal field without timeunit"
and
"should enumerate correct scaleType for nominal field with timeunit"

should be grouped into 

"should enumerate correct scaleType for nominal field"